pkg/util/spanlogger/spanlogger.go

```diff
@@ -94,3 +95,23 @@ func (s *SpanLogger) Error(err error) error {
 	s.Span.LogFields(otlog.Error(err))
 	return err
 }
+
+// triggerError calls s.Error if SpanLogger's log level is set to level.ErrorValue().
+func (s *SpanLogger) triggerError(kvps ...interface{}) {
+	var logAsError = false
+	loggedError := errors.New("unknown error")
+	for i := 0; i < len(kvps) - 1; i += 2 {
+		if kvps[i] == level.Key() && kvps[i+1] == level.ErrorValue() {
```

Comment:
No need to continue the iteration if the level is not error. I think for the purpose of this `triggerError()` function you could just `return` if level != error.